### ISSUE TITLE
Feature planning for v3.0

### ISSUE THREAD

**Issue by Dovyski** [FlixelCommunity/flixel#193]
This issue is aimed to keep track of the features/changes planned to land on Flixel Community v3.0.

| Feature/Change | Author | Complexity |
| --- | --- | --- |
| ~~Moving classes into folders: core (FlxBasic, FlxObject), graphics (FlxSprite, FlxText), etc (#194)~~ Added in ca946ab5b2a60ca7d4c1df13e84d29fb2fb40c01 | @WingEraser | Minor |
| ~~Make Flixel GPU accelerated (#90)~~ WIP in #219 | @Dovyski | **_Major**_ |
| Organize FlxG and FlxU a tad better (#142) [Forum post about it](http://forums.flixel.org/index.php/topic,5675). | @IQAndreas | Minor |
| Add math vector methods to FlxPoint (or create FlxVector) (#188) | @Dovyski | Minor |
| Add more colors to FlxG (or preferably, FlxColor) (#141) | @IQAndreas | Minor |
| ~~Add FlxRandom seeded random generator (#197)~~ Added in 8bad64f6e9bfba12cc4deae8518de44e09cef474 | @IQAndreas | Minor |
| ~~Move static constants on path following from FlxObject to FlxPath (#112)~~ WIP in #218 | @IQAndreas | Minor |
| Merge a part of Flixel Power Tools into code tree (#163) | @IQAndreas | **_Major**_ |
| ~~Rename and add FlxDebugger functions ([discussions](https://github.com/FlixelCommunity/flixel/issues/158#issuecomment-15914186))~~ WIP in #217 | @IQAndreas | Minor |
| ~~Improve plugin system (#121)~~ WIP in 79450d8c6d91a88f67cc8a950bfcb079eb9e44d9 | @WingEraser | _Medium/Major_ |
| Create tutorials for Flixel Community website ([issue #3 on flixelcommunity.github.com](https://github.com/FlixelCommunity/flixelcommunity.github.com/issues/3)) | @Dovyski | Minor |

After adding an entry to the table, please, post a comment discussing it.

**Comment by IQAndreas** [comment]
Nice table! Thanks for filling everything in.

Shall I copy over some of the discussion as well, or should we just continue from where we left off?

**Comment by IQAndreas** [comment]
> Merge a part of Flixel Power Tools into code tree (#163)

As for this one, I was thinking **primarily** of these classes, as they would compliment #141 and #142:
- [FlxColor](https://github.com/photonstorm/Flixel-Power-Tools/blob/master/Test%20Suite/src/org/flixel/plugin/photonstorm/FlxColor.as)
- [FlxMath](https://github.com/photonstorm/Flixel-Power-Tools/blob/master/Test%20Suite/src/org/flixel/plugin/photonstorm/FlxMath.as)

But others can of course come in as well if they fit in better as part of the core rather than plugin.

**Comment by WingEraser** [comment]
If you don't mind, I can take #142 as well while I'm refactoring the classes.
EDIT: also #141

**Comment by IQAndreas** [comment]
> If you don't mind, I can take #142 as well while I'm refactoring the classes.

Aw, I was actually looking forward to doing those. :stuck_out_tongue: If it's alright that I do those, I had some plans on how to group the "deprecated redirect" methods.

But I do think, if we move everything into packages, since it affects so many files, that should be done before any other changes so as to avoid merge conflicts. Do you have an IDE where you just drag and drop the classes to their new locations, or do you need to do that work manually?

**Comment by WingEraser** [comment]
I got FDT4 which should work out. I'll let you handle the #141 and #142. I'm going to start this weekend with a folder structure and put it in a separated issue.

**Comment by IQAndreas** [comment]
Just a quick check first, we already talked about the cons of some of these proposed v2.58 features by email, but is there anyone that is adamantly against any of the changes or feel they should not be used?

**Comment by Dovyski** [comment]
> Shall I copy over some of the discussion as well, or should we just continue from where we left off?

Let's continue from where we left off.

> is there anyone that is adamantly against any of the changes or feel they should not be used?

I'm ok with all changes! Of course I would like to discuss some things, such as which parts of Flixel Power Tools should be merged, which new singletons to create (e.g. `FlxM`, etc), etc, but all that can be done as we move to the implementation of each task.

**Comment by Dovyski** [comment]
I can take #90 to start working on Stage3D support for Flixel. It will take a lot of experimentation and prototipation. Anyone else interested on this particular issue?

**Comment by WingEraser** [comment]
> I can take #90 to start working on Stage3D support for Flixel. It will take a lot of experimentation and prototipation. Anyone else interested on this particular issue? 

I will join, but only for the thinking part. Can't start with this yet until next year, when I got more free time.

**Comment by IQAndreas** [comment]
> Of course I would like to discuss some things, such as which parts of Flixel Power Tools should be merged, which new singletons to create (e.g. FlxM, etc), etc.

Excellent. I have already started on said discussion in https://github.com/FlixelCommunity/flixel/issues/142#issuecomment-29441081, just chime in there. Or should I create a separate issue for `FlxMath` and another for `FlxColor`?

**Comment by IQAndreas** [comment]
On a related topic, now that we have a stable release that is publicly available, shall we change the main (default) branch displayed in the repository to `master`? (currently, it is `dev`)

**Comment by Dovyski** [comment]
> I will join, but only for the thinking part. Can't start with this yet until next year, when I got more free time.

Great! I will post my ideas on #90, we can discuss everything there.

> Or should I create a separate issue for FlxMath and another for FlxColor

Let's use the already open issue, it's easier to follow the discussion.

About the main branch displayed in the repository, I totally agree about using `master`. I thought I was the only one seeing `dev` as default =P

**Comment by FlixelCommunityBot** [comment]
# Flixel v3.00 (CHANGELOG)

TODO: Insert description of this release

#### Repository changes

Change main package from `org.flixel` to just `flixel` ([#194](https://github.com/FlixelCommunity/flixel/issues/194))
Organize Flixel's classes into several, more logically organized packages ([#194](https://github.com/FlixelCommunity/flixel/issues/194))
Rename `FlxAnim` to `FlxAnimation` ([#205](https://github.com/FlixelCommunity/flixel/issues/194))

#### New Features

New plugin system based on `FlxSignals` ([#121](https://github.com/FlixelCommunity/flixel/issues/121))

#### Improvements

Example improvement ([#0](https://github.com/FlixelCommunity/flixel/issues/0))  

#### Performance Improvements

Example performance improvement ([#0](https://github.com/FlixelCommunity/flixel/issues/0))  

#### Bugfixes

Example bug fix ([#0](https://github.com/FlixelCommunity/flixel/issues/0))
